Sources/SwiftSyntaxBuilder/IfStmtConvenienceInitializers.swift

```diff
@@ -0,0 +1,34 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+
+public extension IfStmt {
+  /// A convenience initializer that uses builder closures to express an
+  /// if body, potentially with a second trailing builder closure for an else
+  /// body.
+  init(
```

Comment:
I’m not sure if we have other types that take an optional parameter that’s expressible by a result builder but do you think it might be worth generalizing this case and putting it into the generated code?

Reply:
That's an interesting question, I am not sure whether this would generalize well though. AFAIK Swift doesn't have many if-else-ish constructs with multiple blocks. Aside from that, I don't think we're actively discriminating against multiple closures currently: The else-body in `IfStmt` doesn't seem to be a `CodeBlock`, but a `Syntax`, which also is the reason why an initializer of this form isn't autogenerated (I am not entirely sure why this is the case since I can't think of a case where an `else` wouldn't be followed by a `CodeBlock`, but that is also the reason why we need to perform this wrapping explicitly in this convenience initializer).

Reply:
I see. Looking at `gyb_syntax_support`, the else body could also be an `IfStmt` to represent an `else if` clause. With that in mind, I think this PR makes perfect sense 👍